### DISCUSSION TITLE
Symbol diff Problem

### DISCUSSION THREAD

**Discussion by Gui-Yue** [linuxdeepin/developer-center#4762] [Troubleshooting | 故障排查]
在给kde相关包做升级工作的时候，遇到了symbol差异的问题。以下是涉及的软件包：

[kactivities-kf5](https://build.deepin.com/package/show/deepin:CI:topics:kde-relate/kactivities-kf5)
![kactivities-kf5](https://github.com/linuxdeepin/developer-center/assets/78520005/238f43bd-8df3-461a-abe2-cbcb6c5a4ce2)
[kconfigwidgets](https://build.deepin.com/package/show/deepin:CI:topics:kde-relate/kconfigwidgets)
![kconfigwidget](https://github.com/linuxdeepin/developer-center/assets/78520005/7603ab05-a7f7-4f6f-a01f-52b6ba6c7540)
[kio](https://build.deepin.com/package/show/deepin:CI:topics:kde-relate/kio)
![kio](https://github.com/linuxdeepin/developer-center/assets/78520005/467136f0-f8f3-4f73-bf9f-6f09f002e72f)
我目前能想到的解决办法有三个：
1、更新该三个软件包的symbol避免symbol差异。
2、修改这三个软件包的rules文件跳过symbol的检查。
3、更新obs上的编译器（此方案影响较大，需要进一步讨论分析是否可行）

请问上述的symbol差异的问题应该以何种方式解决？

**Comment by Zeno-sole** [comment]
我比较推荐第一种解决方案 ， 更新编译工具链短时间肯定是无法推进的

**Reply by Gui-Yue** [reply]
如果修改symbol文件，是否需要将改动概述写到changelog中以视标记，如果不需要，那么应该以何种方式来注明该软件包symbol文件修改过？

**Comment by RevySR** [comment]
> 我比较推荐第一种解决方案 ， 更新编译工具链短时间肯定是无法推进的

第一种方案会和riscv架构冲突 不建议这样操作

**Comment by Zeno-sole** [comment]
不这样做就和主线冲突



---Original---
From: "Han ***@***.***&gt;
Date: Fri, Jun 23, 2023 22:26 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [linuxdeepin/developer-center] Symbol diff Problem (Discussion#4762)




  
我比较推荐第一种解决方案 ， 更新编译工具链短时间肯定是无法推进的
  
第一种方案会和riscv架构冲突 不建议这样操作
